src/zkevm_specs/evm_circuit/execution/error_code_store.py

```diff
@@ -0,0 +1,40 @@
+from zkevm_specs.evm_circuit.table import RW
+from zkevm_specs.util.param import (
+    GAS_COST_CODE_DEPOSIT,
+    MAX_CODE_SIZE,
+    N_BYTES_MEMORY_ADDRESS,
+    N_BYTES_STACK,
+)
+from ...util import FQ
+from ..instruction import Instruction
+from ..opcode import Opcode
+from ...util import N_BYTES_GAS
+
+
+def error_code_store(instruction: Instruction):
+    # retrieve op code associated to oog constant error
+    opcode = instruction.opcode_lookup(True)
+    instruction.constrain_equal(opcode, Opcode.RETURN)
+
+    # the call must be coming from CREATE or CREATE2
+    instruction.constrain_equal(FQ(instruction.curr.is_create), FQ(1))
+
+    # pop length (which is the bytecode size) from stack
+    return_length_word = instruction.stack_lookup(RW.Read, 1)
+    return_length = instruction.word_to_fq(return_length_word, N_BYTES_MEMORY_ADDRESS)
+
```

Comment:
MD document mentions the current call is not to be static call - is this constraint missing here?

Reply:
Nice catch! fixed in [1365730](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/475/commits/1365730d5f43234dbf650a3ef918c7cee3251220)